src/core/meshcop/secure_transport.cpp

```diff
@@ -159,18 +161,37 @@ Error SecureTransport::Open(ReceiveHandler aReceiveHandler, ConnectedHandler aCo
 
     mConnectedCallback.Set(aConnectedHandler, aContext);
     mReceiveCallback.Set(aReceiveHandler, aContext);
+    mShouldLimitConnectionAttempts = false;
+
     SetState(kStateOpen);
 
 exit:
     return error;
 }
 
+void SecureTransport::SetMaxConnectionAttempts(uint16_t aMaxAttempts, AutoCloseCallback aCallback, void *aContext)
+{
+    VerifyOrExit(mState != kStateClosed);
```

Comment:
Is it better to require setting these parameters only when `mState == kStateClosed`? Are there use cases where we would want to support multiple calls to `SetMaxConnectionAttempts()` while in other states?

Reply:
This can also work.

I was considering this as a limit we can apply while socket is open and will be applied once (i.e., once socket is closed and if we re-open it won't apply the limit again).

Let me change it so that we can only set it when socket is closed and then the limit will apply to socket until `SetMaxConnectionAttempst()` is called again.